src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala

```diff
@@ -16,22 +18,32 @@ import scala.concurrent.{ExecutionContext, Future}
 /**
   * Created by davidan on 10/7/16.
   */
-trait RestJsonClient {
-
+trait RestJsonClient extends FireCloudRequestBuilding {
   implicit val system: ActorSystem
   implicit val executionContext: ExecutionContext
 
-  def userAuthedRequest(req: HttpRequest, compressed: Boolean = false)(implicit userInfo: WithAccessToken): Future[HttpResponse] = {
-    val pipeline = if (compressed) {
-      addCredentials(userInfo.accessToken) ~> addHeader (`Accept-Encoding`(gzip)) ~> sendReceive ~> decode(Gzip)
-    } else {
-      addCredentials(userInfo.accessToken) ~> sendReceive
+  def userAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false)(implicit userInfo: WithAccessToken): Future[HttpResponse] = {
```

Comment:
userAuthedRequest and adminAuthedRequest are identical except for the function to add creds, make an internal function that takes the addCreds function as a param.
```
  def userAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false)(implicit userInfo: WithAccessToken): Future[HttpResponse] = authedRequest(addCredentials(userInfo.accessToken))(req, compressed, useFireCloudHeader)
  def adminAuthedRequest(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false): Future[HttpResponse] = authedRequest(addAdminCredentials)(req, compressed, useFireCloudHeader)

  private def authedRequest(addCreds: RequestTransformer)(req: HttpRequest, compressed: Boolean = false, useFireCloudHeader: Boolean = false)(implicit userInfo: WithAccessToken): Future[HttpResponse] = {
    val pipeline = (compressed, useFireCloudHeader) match {
      case (true, true) => addCreds ~> addFireCloudCredentials ~> addHeader (`Accept-Encoding`(gzip)) ~> sendReceive ~> decode(Gzip)
      case (true, false) => addCreds ~> addHeader (`Accept-Encoding`(gzip)) ~> sendReceive ~> decode(Gzip)
      case (false, true) => addCreds ~> addFireCloudCredentials ~> addHeader (`Accept-Encoding`(gzip)) ~> sendReceive ~> decode(Gzip)
      case _ => addCreds ~> sendReceive
    }
    pipeline(req)
  }
```